File: CS/GridDemo.Wpf/ModuleResources/PrintViewModelBase.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Printing;
using System.Linq;
using System;
using DevExpress.Xpf.DemoBase.Helpers;

namespace GridDemo {
    public abstract class PrintViewModelBase {
        IPrintableControl printableControl;
        protected PrintViewModelBase() {

        }
        public virtual void OnLoaded(IPrintableControl printableControl) {
            this.printableControl = printableControl;
        }
        public void OnUnloaded() {
            var service = this.GetService<IDocumentManagerService>();
            foreach(var document in service.Documents.ToArray()) {
                OnTabClosed(document.Content);
                document.Close();
            }
        }
        public void ShowPreviewInNewTab() {
            Logger.Log("ShowPreviewInNewTab");
            var service = this.GetService<IDocumentManagerService>();
            var link = new PrintableControlLink(printableControl);
            link.CreateDocument(true);
            var doc = service.CreateDocument(link);
            doc.Title = GetTitle();
            doc.DestroyOnClose = true;
            doc.Show();
        }
        public void OnTabClosed(object tabContent) {
            ((PrintableControlLink)tabContent).Dispose();
        }
        protected abstract string GetTitle();
    }
}

File: VB/GridDemo.Wpf/ModuleResources/PrintViewModelBase.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.Printing
Imports System.Linq
Imports System
Imports DevExpress.Xpf.DemoBase.Helpers

Namespace GridDemo
    Public MustInherit Class PrintViewModelBase
        Private printableControl As IPrintableControl
        Protected Sub New()

        End Sub
        Public Overridable Sub OnLoaded(ByVal printableControl As IPrintableControl)
            Me.printableControl = printableControl
        End Sub
        Public Sub OnUnloaded()
            Dim service = Me.GetService(Of IDocumentManagerService)()
            For Each document In service.Documents.ToArray()
                OnTabClosed(document.Content)
                document.Close()
            Next document
        End Sub
        Public Sub ShowPreviewInNewTab()
            Logger.Log("ShowPreviewInNewTab")
            Dim service = Me.GetService(Of IDocumentManagerService)()
            Dim link = New PrintableControlLink(printableControl)
            link.CreateDocument(True)
            Dim doc = service.CreateDocument(link)
            doc.Title = GetTitle()
            doc.DestroyOnClose = True
            doc.Show()
        End Sub
        Public Sub OnTabClosed(ByVal tabContent As Object)
            DirectCast(tabContent, PrintableControlLink).Dispose()
        End Sub
        Protected MustOverride Function GetTitle() As String
    End Class
End Namespace
